autosklearn/data/validation.py

```diff
@@ -15,8 +17,8 @@
 
 
 def convert_if_sparse(
-    y: typing.Union[SUPPORTED_TARGET_TYPES, spmatrix]
-) -> SUPPORTED_TARGET_TYPES:
+    y: SUPPORTED_TARGET_TYPES
+) -> Union[np.ndarray, List, pd.DataFrame, pd.Series]:
```

Comment:
It definitely can, however I think this requires a proper discussion, perhaps during a meeting or roadmap.

* Do we type the function as it can support? (wide, broadly useable, more code)
* Do we type the function as it expects to be used? (narrow, less useablea, less code)

Reply:
Discussion outcome: narrow typing

Will this be changed in this PR or in an upcoming (might be better as this PR is already quite large).

Reply:
Upcoming PR